### ISSUE TITLE
Is there any compact implementation of Keccak?

### ISSUE THREAD

**Issue by MaiaVictor** [gvanas/KeccakCodePackage#40]
[This file](https://github.com/gvanas/KeccakCodePackage/blob/master/Standalone/CompactFIPS202/Python/CompactFIPS202.py) implements SHA3. I wonder if there is a similar thing for Keccak (as per the winning entry to the SHA-3 contest by Bertoni et al. [2017], rather than later releases).

**Comment by MaiaVictor** [comment]
Alternatively, if you could point the exact difference between the two versions, so I could modify that file accordingly.

**Comment by gvanas** [comment]
Hi Victor,

TL;DR: just call `Keccak(rate, capacity, inputBytes, 0x01, outputByteLen)`.

The FIPS 202 instances (so SHA3-* and SHAKE*) are just a layer on top of (unchanged) Keccak, that adds a short suffix to the input for domain separation. The Python code you are referring to follows this layering and implements Keccak with any input. If you don't want to add any suffix (e.g., as in Keccak as used in Ethereum), just called the `Keccak()` function with `delimitedSuffix=0x01`. The meaning of the `delimitedSuffix` value can be found [here](https://github.com/gvanas/KeccakCodePackage/blob/master/doc/KeccakSponge-documentation.h#L43).

I hope this helps.

Kind regards,
Gilles

**Comment by MaiaVictor** [comment]
Yes, a lot. Thank you so much! :)
